module-options.nix

```
@@ -84,6 +84,11 @@ in
 
     # Outputs
     build = {
+      devShell = mkOption {
+        description = "The development shell with treefmt and its underlying programs";
+        type = types.package;
+        default = pkgs.mkShell { buildInputs = [ config.build.wrapper ] ++ (lib.attrValues config.build.programs); };
```

```suggestion
        default = pkgs.mkShell { nativeBuildInputs = [ config.build.wrapper ] ++ (lib.attrValues config.build.programs); };
```

(So you get shell completion, as @roberth says)

You probably also want to set `readonly = true;` in this option.